src/algorithms/expected_consensus.md

```diff
@@ -197,11 +220,12 @@ Due to the existence of potential forks in EC, a miner can try to unduly influen
 
 This is detectable when a given miner submits two blocks that satisfy any of the following "consensus faults":
 
-- (1) `double-fork mining fault`: two blocks mined off of the same Tipset at different epochs (i.e. with different `ElectionProof`s generated from the same input ticket).
-- (2) `parent grinding fault`: one block's parent is a Tipset that provably should have included a given block but does not. While it cannot be proven that a missing block was willfully omitted in general (i.e. network latency could simply mean the miner did not receive a particular block), it can when a miner has successfully mined a block two epochs in a row and omitted one. That is, this condition should be evoked when a miner omits their own prior block. When a miner's block at epoch e + 1 references a Tipset that does not include the block they mined at e both blocks can be submitted to prove this fault.
+- (1) `double-fork mining fault`: two blocks mined at the same epoch.
+- (2) `same-fork double-mining fault`: two blocks mined off of the same Tipset at different epochs (i.e. with different `ElectionProof`s generated from the same input ticket).
+- (3) `parent grinding fault`: one block's parent is a Tipset that provably should have included a given block but does not. While it cannot be proven that a missing block was willfully omitted in general (i.e. network latency could simply mean the miner did not receive a particular block), it can when a miner has successfully mined a block two epochs in a row and omitted one. That is, this condition should be evoked when a miner omits their own prior block. When a miner's block at epoch e + 1 references a Tipset that does not include the block they mined at e both blocks can be submitted to prove this fault.
 
```

Comment:
@ZenGround0 would love thoughts on above stated conditions if you have!

Reply:
What exactly means by 
*(1) `double-fork mining fault`: two blocks mined at the same epoch.*  ?

For example, in the following case, are [n+2] and [n+2'] at the same epoch? 
[n] <--- [n+1] <--- [n+2]
 &nbsp;&nbsp; \\
&nbsp;&nbsp;&nbsp;&nbsp;\\---- [n+1'] <---[n+2']

Reply:
yes!

Reply:
a dot diagram may be useful there -- here:

```dot
digraph G {
  rankdir=RL;
  node [shape=box, style="filled", color="lightgreen"]

  B1 [label="B1\nMinerAny"]
  B2 [label="B2\nMinerA",   color=pink]
  B3 [label="B3\nMinerA",   color=pink]
  B4 [label="B4\nMinerB",   color=pink]
  B5 [label="B5\nMinerB",   color=pink]
  B6 [label="B6\nMinerC"]
  B7 [label="B7\nMinerC"]

  B6 -> B3
  B7 -> B4

  B4 -> B2 -> B1
  B5 -> B3 -> B1
}
```

looks like

![image](https://user-images.githubusercontent.com/138401/68203958-d28c9680-ff7b-11e9-8c6f-e695b550c0f1.png)